src/main/java/forum/rest/ThreadRestController.java

```diff
@@ -0,0 +1,39 @@
+package forum.rest;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import forum.entity.Category;
+import forum.entity.Thread;
+import forum.service.ThreadService;
+
+@RestController
+@RequestMapping("/api")
+public class ThreadRestController {
+
+	@Autowired
+	private ThreadService threadService;
+
+	@Autowired
+	private CategoryRestController categoryController;
+
+	@PostMapping("/threads")
+	public Thread addThread(@RequestBody Thread theThread) {
+
+		theThread.setId(0);
+
+		threadService.saveThread(theThread);
+
+		theThread.setCategory(categoryController.getCategory(theThread.getCategory().getId()));
```

Comment:
This controller should call the `threadService` and get back from there the thing we send back to the client. So the logic here doesnt really belong to this file

Reply:
Moved the logic to the ThreadService